app/src/main/java/org/oppia/android/app/customview/VerticalDashedLineView.kt

```diff
@@ -0,0 +1,67 @@
+package org.oppia.android.app.customview
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.DashPathEffect
+import android.graphics.Paint
+import android.util.AttributeSet
+import android.util.TypedValue
+import android.view.View
+import androidx.core.content.ContextCompat
+import org.oppia.android.R
+
+private const val DASH_GAP_IN_DP = 16
+private const val DASH_LENGTH_IN_DP = 32
+private const val DASH_THICKNESS_IN_DP = 6
+
+/**
+ * Custom view to represent chapter progress in [StoryActivity].
+ *
+ * Reference: https://stackoverflow.com/a/27054463
+ */
+class VerticalDashedLineView @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null
+) : View(context, attrs) {
+  private val paint: Paint
+  private lateinit var canvas: Canvas
+
+  init {
+    val dashGap: Float = dpToPx(DASH_GAP_IN_DP)
+    val dashLength: Float = dpToPx(DASH_LENGTH_IN_DP)
+    val dashThickness: Float = dpToPx(DASH_THICKNESS_IN_DP)
```

Comment:
Optional, but we could alternatively define these as custom attributes or binding adapter values to make the view more customizable.

Reply:
Actually that was the original solution in the reference too and we can still do that custom attributes solution but I don't see any pros of that I am more inclined towards keeping everything for this UI in one single file so that it does not get changed unintentionally.

Reply:
Okay, seems reasonable.